src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/hooks/useFlagVariation.tsx

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import { CFVariationColors } from '@cf/constants'
+import type { Variation } from 'services/cf'
+
+interface UseFlagVariationProps {
+  flagVariations: Variation[]
+}
+
+interface UseFlagVariationReturn {
+  getVariationColorById: (variationIdentifier: string) => string
+  getVariationColorByName: (variationName: string) => string
+  getVariationNameById: (variationIdentifier: string) => string
+}
+
+const useFlagVariation = ({ flagVariations }: UseFlagVariationProps): UseFlagVariationReturn => {
+  const variationColorMap: {
+    [key: string]: string
+  } = {}
+
+  flagVariations.forEach((variation, variationIndex) => {
+    variationColorMap[variation.identifier] = CFVariationColors[variationIndex]
+  })
```

Comment:
You could combine the declaration and instantiation using a `reduce`
```suggestion
  const variationColorMap = flagVariations.reduce<Record<string, string>>((map, variation, variationIndex) => ({
    ...map, 
    [variation.identifier]: CFVariationColors[variationIndex]
  }), {})
```

Reply:
nice 👍